docs/ossm/ossm2-migration/cluster-wide/README.md

```diff
@@ -0,0 +1,384 @@
+     <!--TODO: add a link when the doc is ready https://issues.redhat.com/browse/OSSM-8520-->
+  1. Ensure that the `discoverySelectors` defined in your OpenShift Service Mesh 3.0 `Istio` resource will match the namespaces that make up your OpenShift Service Mesh 2.6 mesh. You may need to add additional labels onto your OpenShift Service Mesh 2.6 application namespaces to ensure that they are captured by your OpenShift Service Mesh 3.0 `Istio` `discoverySelectors`. See [Scoping the service mesh with DiscoverySelectors](../../create-mesh/README.md)
+- only one control plane will try to inject a side car
+
+  This can be achieved by correct use of injection labels. Please see [Installing the Sidecar](../../injection/README.md) for details.
+  > **_NOTE:_** Due to specific behavior of OpenShift Service Mesh 2.6 it's necessary to disable 2.6 injector when migrating the data plane namespace. We will use `maistra.io/ignore-namespace: "true"` label in this guide.
+
+Apart from the conditions above, it's necessary to decide which injection labels will be used. See [Installing the Sidecar](../../injection/README.md) explaining relation between Istio revisions and injection labels.
+
+Once it's clear how the `istio.io/rev` and `istio-injection` labels work in OpenShift Service Mesh 3, it's also necessary to revisit your OpenShift Service Mesh 2.6 installation and understand consequences of different injection configurations. Typically, following configurations might be used:
+
+- by default the `spec.memberSelectors` in your `ServiceMeshMemberRoll` is configured to match `istio-injection=enabled` label and all of your 2.6 data plane namespaces are already labeled with `istio-injection=enabled`
+
+    With this configuration, you will be able to keep the `istio-injection=enabled` on your data plane namespaces during the migration.
+- `spec.memberSelectors` in your `ServiceMeshMemberRoll` is not configured to match `istio-injection=enabled` and your 2.6 data plane namespaces are using some other label
+
+    With this configuration, it will be necessary to add either the `istio.io/rev` or `istio-injection` label during the migration. The label defined in the `spec.memberSelectors` in your `ServiceMeshMemberRoll` will have no effect on injection in OpenShift Service Mesh 3
+- [adding projects using label selectors](https://docs.openshift.com/container-platform/4.16/service_mesh/v2x/ossm-create-mesh.html#ossm-about-adding-projects-using-label-selectors_ossm-create-mesh) feature is not used at all and all projects were added to the mesh manually by creating `ServiceMeshMember`
+
+    With this configuration, it will be necessary to add either the `istio.io/rev` or `istio-injection` label during the migration.
+
+Based on your 2.6 configuration listed above, you should follow one of the procedures bellow.
```

Comment:
I must admit, reading the above, I'm not clear on which procedure to choose, and the scenario in which the 3rd path would cause down time. It looks like the options are:
1. I am using `istio.io/rev` labels, and want to use a revision based upgrade procedure and to continue using `istio.io/rev` labels.
2. I am using `istio-injection=enabled` labels and want to use a revision based upgrade procedure to ensure zero downtime, but then want to move back to just using `istio-injection=enabled` labels.
3. I am using `istio-injection=enabled`, and don't want to use`istio.io/rev` labels as part of the upgrade procedure. It says that this creates traffic disruption risk, but it's not clear how that risk occurs.

Reply:
I will expand this section.